app/components/certificate/certificate-display.tsx

```diff
@@ -0,0 +1,111 @@
+import {
+  Flex,
+  Box,
+  HStack,
+  Text,
+  Heading,
+  Tooltip,
+  IconButton,
+  Accordion,
+  AccordionItem,
+  AccordionButton,
+  AccordionIcon,
+  AccordionPanel,
+  useToast,
+} from '@chakra-ui/react';
+import { DownloadIcon, CopyIcon } from '@chakra-ui/icons';
+import type { Certificate } from '@prisma/client';
+
+interface Props {
+  cert: Certificate;
+  priv: Boolean;
+}
+
+export default function CertificateDisplay(props: Props) {
+  const toast = useToast();
+
+  function onCopy() {
+    let key = props.cert.certificate;
+    let val = 'Public Key';
+
+    if (props.priv) {
+      key = props.cert.privateKey;
+      val = 'Private Key';
+    }
+
+    navigator.clipboard.writeText(key);
+    toast({
+      title: `${val} was copied to clipboard`,
+      position: 'bottom-right',
+      status: 'success',
+    });
+  }
+
+  return (
+    <Flex flexDirection="column" gap="5">
+      <Flex flexDirection="column" gap="4">
+        <HStack gap="4">
+          <Heading as="h4" size="sm">
+            {props.priv ? 'Private Key' : 'Public Key'}
+          </Heading>
+          <Tooltip label="Copy Public Key">
```

Comment:
This `public` vs. `private` logic should get figured out via props vs hard-coding like this and below.

Reply:
I decided to create a `switch` and 3 new variables that will be used for the `key`, `currentState` and `Description`. This should remove any need for hard coding, because the output is depended on the current state of the `isPrivate` props